src/v/kafka/server/request_context.h

```diff
@@ -146,12 +146,11 @@ class request_context {
     ss::future<response_ptr> respond(ResponseType r) {
         vlog(
           klog.trace,
-          "[{}:{}] sending {}:{} response {}",
+          "[{}:{}] sending {}:{}",
           _conn->client_host(),
           _conn->client_port(),
           ResponseType::api_type::key,
-          ResponseType::api_type::name,
-          r);
```

Comment:
I had this thought too, but decided to err on the conservative side with what to log.

My concern is that, while this PR addresses handling sensitive system information (credentials, passwords, etc.), there are still gaps in redaction with regards to user data (e.g. topic names). These aren't necessarily sensitive, but they're data that could eventually be worth scrubbing, since user data may include PII.

The discussion begets some potential scope creep to what we're looking to redact, but I think it's an important to discussion to have while we're considering what to keep. How do you feel about also redacting any request/response that contains any user data?

Reply:
i think we should bring this up on slack for discussion. not being able to log topic names seems quite problematic. plus, topic names are logged all over the code base, not just in kafka.

Reply:
I left this in, and went the route of redacting individual message types.

Also, for posterity, there was a brief discussion on what we should consider redacting right at this moment, and topic names did not make the cut.

That said, the updated PR includes redaction of more things, like fetch response bodies.